_posts/2020-05-08-Rails-6-1-beta1-release.markdown

```diff
@@ -0,0 +1,58 @@
+categories: releases
+author: eileencodes
+published: true
+date: 2020-05-08 10:00:00 -05:00
+---
+The first beta for Rails 6.1 has been released and wow does it have a lot of great stuff! We've been hard at work these past few months implementing improvements to multiple databases, adding support for destroying associations in jobs instead of in memory, turning errors into objects, and so much more.
+
+It's amazing how Rails has grown over the years and while we have some improvements to make to the [onboarding process](https://weblog.rubyonrails.org/2020/5/7/A-May-of-WTFs/), Rails has never been better. The features in this release focus on adding the functionality you need to keep your application up and running for years to come.
+
+Let's look at some of the new functionality:
+
+## Horizontal Sharding
+
+Rails 6.0 provided the ability to functionally partition (multiple partitions, different schemas) your database but wasn't able to support horizontal sharding (same schema, multiple partitions). Rails wasn't able to support horizonal sharding because models in Active Record could only have one connection per-role per-class. This is now fixed and [Eileen M. Uchitelle](https://github.com/eileencodes) worked with [John Crepezzi](https://github.com/seejohnrun) from GitHub to build out this functionality. Check out the [PR](https://github.com/rails/rails/pull/38531)!
+
+## Multi-DB Improvements
+
+In addition to adding horizontal sharding support we added tons of new functionality and improved a lot of internals for multiple databases. [Kyle Thompson](https://github.com/kylekthompson) added support for [database namespaced tasks](https://github.com/rails/rails/pull/38449) like `db:schema:dump:namespace`, `db:schema:load:namespace`, `db:structure:dump:namespace`, and `db:structure:load`. [Jean Boussier](https://github.com/casperisfine) from Shopify improved [connection pool management](https://github.com/rails/rails/pull/37296). [John Crepezzi](https://github.com/seejohnrun) and [Eileen M. Uchitelle](https://github.com/eileencodes) worked on implementing [many](https://github.com/rails/rails/pull/37185) [improvements](https://github.com/rails/rails/pull/37279) [to](https://github.com/rails/rails/pull/38536) [database](https://github.com/rails/rails/pull/38256) [configurations](https://github.com/rails/rails/pull/38029).
+
+## Strict Loading Associations
+
+In addition to all the database and connection management improvements, [Aaron Patterson](https://github.com/tenderlove) and [Eileen M. Uchitelle](https://github.com/eileencodes) added support for [strict loading associations](https://github.com/rails/rails/pull/37400). With this feature you can ensure that all your associations are loaded eagerly and stop N+1's before they happen. [Kevin Deisz](https://github.com/kddeisz) added additional support to [association declarations](https://github.com/rails/rails/pull/38541).
```

Comment:
Should we mention recently added [strict_loading_by_default](https://github.com/rails/rails/pull/39491) feature in this section?